### PR TITLE
week7-assignment

### PR DESCRIPTION
**Pull request by raginimangla**
Added API reference topic - [Create a task](#tasks-create-task) and tutorial topic - [Get tasks for a user](tutorials/get-tasks-for-a-user).

### CODE REVIEW

File: docs/api/tasks-create-task.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Request body
+
+In the request body, specify a JSON representation of the [`task`](task) object. The following table lists the properties that are required when you create a task. 
+
+| Property | Description | Type | Required | Notes |
+| -------------- | ------ | ------------ |------------ |------------ |
+| title| The name of the task.| string | Required |   |
+| description | A short description of the task. | string | Required |  |
+| due-date | The due date of the task. | datetime | Required |  |
+
```
Comment (rbwatson) [inline docs/api/tasks-create-task.md:33]: What about `warning` and `user_id`?

File: docs/api/tasks-create-task.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+The POST body should look something like this. You can change the values of each property as you’d like.
+
+```js
+[
+    {
+        "title": "Assignment 7",
+        "description": "Submit week 7 assignment",
+        "due_date": "2024-05-14T23:59",
+    }
```
Comment (rbwatson) [inline docs/api/tasks-create-task.md:45]: What about `warning` and `user_id`?

File: docs/api/tasks-create-task.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+The following example shows the response. Note that the task title should be the same as you used in your **Request body** and the response should include the new task's id. The task's id is automatically generated when the task is created.
+
+```js
+[
+    {
+        "id": "4097",
+        "title": "Week 7 assignment",
+        "description": "Submit week 7 assignment",
+        "due_date": "2024-05-14T23:59"
+    }
```
Comment (rbwatson) [inline docs/api/tasks-create-task.md:59]: What about `warning` and `user_id`?

File: docs/tutorials/get-tasks-for-a-user [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        Replace `<user_id>` with the ID of the user for whom you want to retrieve tasks.
+    - **Headers**:
+        - `Content-Type: application/json`
+
+1. Click  **Send** to make the request.
+1. Review the response body, which should contain an array of tasks associated with the specified user. Each task object should include details such as `id`, `title`, `description`, `due_date`, etc.
+
+    ```json
+    [
+        {
```
Comment (rbwatson) [inline docs/tutorials/get-tasks-for-a-user:42]: The tasks in this response have different `user_id` values. That seems odd.

File: docs/tutorials/get-tasks-for-a-user [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    ```shell
+    cd <your-github-workspace>/to-do-service/api
+    json-server -w to-do-db-source.json
+    ```
+
+1. Open the Postman app on your desktop.
+1. Create a new request with the following values:
+    - **METHOD**: GET
+    - **URL**: `{{base_url}}/tasks?user_id=<user_id>`
```
Comment (rbwatson) [inline docs/tutorials/get-tasks-for-a-user:32]:
```suggestion
    - **URL**: `{{base_url}}/tasks?user_id=1`
```
You should use a specific `user_id` in the tutorial so they get a response that matches your documentation. You can then explain that they can change the `user_id` value as needed in future calls.

Also the double braces filter out the text between them. 
See `docs/tutorials/get-a-user-by-name.md` for examples of markup that makes them visible, again.

File: docs/tutorials/get-tasks-for-a-user [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        "user_id": 2,
+        "title": "Oil change",
+        "description": "5K auto service",
+        "due_date": "2024-03-10T09:00",
+        "warning": "-60",
+        "id": "3"
+        }
+    ]
+    ```
+
```
Comment (rbwatson) [inline docs/tutorials/get-tasks-for-a-user:61]:
```suggestion

## Next steps

```
To separate the tutorial from the next steps.

**Review by rbwatson — CHANGES_REQUESTED** [review]
Good topics! Just some questions in the files to resolve.

But your feature branch is out of sync, so you'll need to:
1. Sync your fork
2. Update the main branch on your clone
3. Rebase your feature branch, which might require that you resolve some merge conflicts and review the changes in your editor during the rebase. 
4. Force push the updated feature branch to your fork.
5. Build the docs from your feature branch and review them to make sure they look correct.
6. Update the files to address the comments, and commit those changes.
7. Squash your new commits in the feature branch and force push the branch to your fork.

Then your pull request will be ready to merge.

Thanks!

**Comment by rbwatson** [comment]
If this pull request was replaced by #74, please close this one.

**Comment by raginimangla** [comment]
closing this as new pr #74 was created

**Pull request closed without merging**